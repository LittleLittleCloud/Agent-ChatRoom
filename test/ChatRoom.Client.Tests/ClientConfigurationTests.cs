using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.Json;
using System.Threading.Tasks;
using ApprovalTests;
using ApprovalTests.Namers;
using ApprovalTests.Reporters;
using ChatRoom.SDK;
using FluentAssertions;
using Json.Schema;
using Json.Schema.Generation;
using Xunit;

namespace ChatRoom.Client.Tests;

public class ClientConfigurationTests
{
    [Fact]
    [UseReporter(typeof(DiffReporter))]
    [UseApprovalSubdirectory("ApprovalTests")]
    public void VerifyConfigurationSchema()
    {
        var schema = new JsonSchemaBuilder()
            .FromType<ChatRoomServerConfiguration>()
            .Build();

        var json = JsonSerializer.Serialize(schema, new JsonSerializerOptions { WriteIndented = true });

        var schemaFileName = "client_configuration_schema.json";
        var schemaFilePath = Path.Join("Schema", schemaFileName);
        var schemaFile = File.ReadAllText(schemaFilePath);

        Approvals.Verify(json);
        schemaFile.Should().BeEquivalentTo(json);
    }
}
