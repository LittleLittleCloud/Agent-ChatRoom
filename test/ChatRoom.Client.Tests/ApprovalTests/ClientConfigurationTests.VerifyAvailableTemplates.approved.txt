templates

templates[0] = {
    "$schema": "chatroom_configuration_schema.json",
    "channel_config": {
        "openai_config": {
            "llm_type": "OpenAI",
            "openai_api_key": "<your-api-key>",
            "openai_model_id": "gpt-3.5-turbo"
        }
    },
    "server_config": {
        "urls": "http://localhost:51234;https://localhost:51235"
    }
}

templates[1] = {
    "$schema": "chatroom_configuration_schema.json",
    "channel_config": {
        "openai_config": {
            "llm_type": "OpenAI",
            "openai_api_key": "<your-api-key>",
            "openai_model_id": "gpt-3.5-turbo"
        }
    },
    "server_config": {
        "urls": "http://localhost:51234;https://localhost:51235"
    },
    "chatroom_openai_configuration": {
        "agents": [
            {
                "name": "gpt3.5",
                "system_message": "You are a helpful AI assistant",
                "llm_config": {
                    "llm_type": "OpenAI",
                    "openai_api_key": "<your-api-key>",
                    "openai_model_id": "gpt-3.5-turbo"
                }
            },
            {
                "name": "gpt4",
                "system_message": "You are a helpful AI assistant",
                "llm_config": {
                    "llm_type": "OpenAI",
                    "openai_api_key": "<your-api-key>",
                    "openai_model_id": "gpt-4-turbo"
                }
            },
            {
                "name": "llama3",
                "llm_config": {
                    "llm_type": "ThirdParty",
                    "third_party_llm_endpoint": "http://localhost:51234",
                    "third_party_llm_model_id": "llama3:latest"
                }
            }
        ]
    }
}

templates[2] = {
    "$schema": "chatroom_configuration_schema.json",
    "channel_config": {
        "openai_config": {
            "llm_type": "OpenAI",
            "openai_api_key": "<your-api-key>",
            "openai_model_id": "gpt-3.5-turbo"
        }
    },
    "server_config": {
        "urls": "http://localhost:51234;https://localhost:51235"
    },
    "chatroom_powershell_configuration": {
        "gpt": {
            "CurrentWorkingDirectory": ".",
            "openai_config": {
                "llm_type": "OpenAI",
                "openai_api_key": "<your-api-key>",
                "openai_model_id": "gpt-3.5-turbo"
            }
        },
        "runner": {
            "last_n_message": 3
        }
    }
}

templates[3] = {
    "$schema": "chatroom_configuration_schema.json",
    "channel_config": {
        "openai_config": {
            "llm_type": "OpenAI",
            "openai_api_key": "<your-api-key>",
            "openai_model_id": "gpt-3.5-turbo"
        }
    },
    "server_config": {
        "urls": "http://localhost:51234;https://localhost:51235"
    },
    "chatroom_github_configuration": {
        "github_repo_name": "<your-repo-name>",
        "github_repo_owner": "<your-repo-owner>",
        "github_token": "<your-gh-token>",
        "issue_helper": {
            "openai_config": {
                "llm_type": "OpenAI",
                "openai_api_key": "<your-api-key>",
                "openai_model_id": "gpt-3.5-turbo"
            }
        }
    }
}

templates[4] = {
    "$schema": "chatroom_configuration_schema.json",
    "channel_config": {
        "openai_config": {
            "llm_type": "OpenAI",
            "openai_api_key": "<your-api-key>",
            "openai_model_id": "gpt-3.5-turbo"
        }
    },
    "server_config": {
        "urls": "http://localhost:51234;https://localhost:51235"
    },
    "chatroom_websearch_configuration": {
        "bing_search_config": {
            "bing_api_key": "<your-api-key>",
            "openai_config": {
                "llm_type": "OpenAI",
                "openai_api_key": "<your-api-key>",
                "openai_model_id": "gpt-3.5-turbo"
            }
        },
        "google_search_config": {
            "google_api_key": "<your-api-key>",
            "google_custom_search_engine_id": "<your-cse-id>",
            "openai_config": {
                "llm_type": "OpenAI",
                "openai_api_key": "<your-api-key>",
                "openai_model_id": "gpt-3.5-turbo"
            }
        }
    }
}

templates[5] = {
    "$schema": "chatroom_configuration_schema.json",
    "channel_config": {
        "openai_config": {
            "llm_type": "OpenAI",
            "openai_api_key": "<your-api-key>",
            "openai_model_id": "gpt-3.5-turbo"
        }
    },
    "server_config": {
        "urls": "http://localhost:51234;https://localhost:51235"
    },
    "chatroom_planner_configuration": {
        "react_planner_configuration": {
            "openai_config": {
                "llm_type": "OpenAI",
                "openai_api_key": "<your-api-key>",
                "openai_model_id": "gpt-4o"
            }
        }
    }
}

templates[6] = {
    "$schema": "chatroom_configuration_schema.json",
    "channel_config": {
        "openai_config": {
            "llm_type": "OpenAI",
            "openai_api_key": "<your-api-key>",
            "openai_model_id": "gpt-3.5-turbo"
        }
    },
    "server_config": {
        "urls": "http://localhost:51234;https://localhost:51235"
    },
    "chatroom_github_configuration": {
        "github_repo_name": "<your-repo-name>",
        "github_repo_owner": "<your-repo-owner>",
        "github_token": "<your-gh-token>",
        "issue_helper": {
            "openai_config": {
                "llm_type": "OpenAI",
                "openai_api_key": "<your-api-key>",
                "openai_model_id": "gpt-3.5-turbo"
            }
        }
    },
    "chatroom_planner_configuration": {
        "react_planner_configuration": {
            "openai_config": {
                "llm_type": "OpenAI",
                "openai_api_key": "<your-api-key>",
                "openai_model_id": "gpt-4o"
            }
        }
    },
    "chatroom_openai_configuration": {
        "agents": [
            {
                "name": "gpt3.5",
                "system_message": "You are a helpful AI assistant",
                "llm_config": {
                    "llm_type": "OpenAI",
                    "openai_api_key": "<your-api-key>",
                    "openai_model_id": "gpt-3.5-turbo"
                }
            },
            {
                "name": "gpt4",
                "system_message": "You are a helpful AI assistant",
                "llm_config": {
                    "llm_type": "OpenAI",
                    "openai_api_key": "<your-api-key>",
                    "openai_model_id": "gpt-4-turbo"
                }
            },
            {
                "name": "llama3",
                "llm_config": {
                    "llm_type": "ThirdParty",
                    "third_party_llm_endpoint": "http://localhost:51234",
                    "third_party_llm_model_id": "llama3:latest"
                }
            }
        ]
    },
    "chatroom_powershell_configuration": {
        "gpt": {
            "CurrentWorkingDirectory": ".",
            "openai_config": {
                "llm_type": "OpenAI",
                "openai_api_key": "<your-api-key>",
                "openai_model_id": "gpt-3.5-turbo"
            }
        },
        "runner": {
            "last_n_message": 3
        }
    },
    "chatroom_websearch_configuration": {
        "bing_search_config": {
            "bing_api_key": "<your-api-key>",
            "openai_config": {
                "llm_type": "OpenAI",
                "openai_api_key": "<your-api-key>",
                "openai_model_id": "gpt-3.5-turbo"
            }
        },
        "google_search_config": {
            "google_api_key": "<your-api-key>",
            "google_custom_search_engine_id": "<your-cse-id>",
            "openai_config": {
                "llm_type": "OpenAI",
                "openai_api_key": "<your-api-key>",
                "openai_model_id": "gpt-3.5-turbo"
            }
        }
    }
}

